framework/test/unit/modules/validators/endpoint.spec.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { InMemoryKVStore, KVStore } from '@liskhq/lisk-db';
+import { Logger } from '../../../../src/logger';
+import { ValidatorsModule } from '../../../../src/modules/validators';
+import { fakeLogger } from '../../../utils/node';
+
+describe('ValidatorsModuleEndpoint', () => {
+	const logger: Logger = fakeLogger;
+	let validatorsModule: ValidatorsModule;
+	const pk = getRandomBytes(48);
+	const address = getRandomBytes(48);
+	const proof = getRandomBytes(48);
+	const getStore1 = jest.fn();
+
+	beforeAll(async () => {
+		validatorsModule = new ValidatorsModule();
+		const subStore = (new InMemoryKVStore() as unknown) as KVStore;
+		await subStore.put(pk, address);
+		const batch = subStore.batch();
+		batch.put(pk, address);
+		await batch.write();
+		getStore1.mockReturnValue(subStore);
+	});
+
+	describe('validateBLSKey', () => {
+		describe('when request data is valid', () => {
+			it('should resolve with appropriate response', async () => {
+				await expect(
+					validatorsModule.endpoint.validateBLSKey({
+						getStore: getStore1,
+						logger,
+						params: {
+							proofOfPossession: proof,
+							blsKey: pk,
+						},
+					}),
+				).resolves.toStrictEqual({ valid: false });
```

Comment:
this should resolve `true` with the test case?
looking at logic I think there should be
- when key already exist
- when key does not exist, but invalid PoP
- when key does not exist, and PoP is valid

Reply:
I think it should resolve `false` since the key already exists in the substore.
As per the LIP 
```
  if there exists an entry in the registered BLS keys substore with storeKey == blsKey:
        return False 
```

Reply:
ah yea that's correct. maybe test description should say that